ymmsl/execution.py

```diff
@@ -90,7 +122,9 @@ def __init__(
             executable: Optional[Path] = None,
             args: Union[str, List[str], None] = None,
             script: Union[str, List[str], None] = None,
-            can_share_resources: bool = True
+            can_share_resources: bool = True,
+            stateful: ImplementationState = ImplementationState.STATEFUL,
+            supports_checkpoint: bool = False
```

Comment:
I'm considering to remove this as I don't see a good use case at this time.

Outcome when checkpointing is enabled:

| yMMSL `Implementation.supports_checkpoint` | Actual implementation supports checkpointing | Outcome |
| --- | --- | --- |
| False | False | Error in `Configuration.check_consistent`, user may edit yMMSL file and end up in next scenario. |
| True | False | Error during the workflow execution by libmuscle |
| False | True | Error in `Configuration.check_consistent`, user **should** edit yMMSL file and end up in next scenario. |
| True | True | This is fine |

I expect the user experience is better when we remove this flag and let the libmuscle Instance crash the workflow early if checkpointing is requested but not implemented. @LourensVeen, what are your thoughts?